src/supervisor-api.ts

```diff
@@ -176,22 +188,29 @@ export class SupervisorAPI {
 		apiTimeout: number,
 	): Promise<void> {
 		const localMode = await this.config.get('localMode');
-		await this.applyListeningRules(localMode || false, port, allowedInterfaces);
+		await this.applyRules(localMode || false, port, allowedInterfaces);
 
 		// Monitor the switching of local mode, and change which interfaces will
 		// be listened to based on that
 		this.config.on('change', changedConfig => {
 			if (changedConfig.localMode != null) {
-				this.applyListeningRules(
+				this.applyRules(
 					changedConfig.localMode || false,
 					port,
 					allowedInterfaces,
 				);
 			}
 		});
 
-		this.server = this.api.listen(port);
-		this.server.timeout = apiTimeout;
+		return new Promise(resolve => {
+			this.server = this.api.listen(port, () => {
```

Comment:
The listen function uses the callback when it is done trying to bind to a port. If an error occurs such as the port is already in use that would not be thrown here. In order to catch an error like that we must use something like

```js
server.on('error', (e) => {
  if (e.code === 'EADDRINUSE') {
    console.log('Address in use, retrying...');
    setTimeout(() => {
      server.close();
      server.listen(PORT, HOST);
    }, 1000);
  }
});
```

[server.listen() docs](https://nodejs.org/api/net.html#net_server_listen)